File: apps/api/app/schemas/watchlist.py
"""
Pydantic schemas for watchlist API endpoints

@module WatchlistSchemas
@version 1.0.0
@since 1.0.0
"""

from pydantic import BaseModel, Field
from typing import Optional


class WatchlistAddRequest(BaseModel):
    """Request to add prospect to watchlist"""
    prospect_id: int = Field(..., description="Prospect ID to add")
    notes: Optional[str] = Field(None, description="Optional notes")
    notify_on_changes: bool = Field(True, description="Enable change notifications")


class WatchlistUpdateNotesRequest(BaseModel):
    """Request to update watchlist notes"""
    notes: str = Field(..., description="Updated notes content")


class WatchlistToggleNotificationsRequest(BaseModel):
    """Request to toggle notifications"""
    enabled: bool = Field(..., description="Enable or disable notifications")


class WatchlistEntry(BaseModel):
    """Watchlist entry response"""
    id: int
    prospect_id: int
    prospect_name: str
    prospect_position: str
    prospect_organization: Optional[str]
    notes: Optional[str]
    added_at: str
    notify_on_changes: bool

    class Config:
        json_schema_extra = {
            "example": {
                "id": 1,
                "prospect_id": 42,
                "prospect_name": "Bobby Witt Jr.",
                "prospect_position": "SS",
                "prospect_organization": "KC",
                "notes": "Great bat speed, solid defense",
                "added_at": "2025-10-03T12:00:00",
                "notify_on_changes": True
            }
        }

File: apps/web/src/lib/api/watchlist.ts
/**
 * API client for watchlist operations
 *
 * @module WatchlistAPI
 * @version 1.0.0
 * @since 1.0.0
 */

import { apiClient } from './client';

export interface WatchlistEntry {
  id: number;
  prospect_id: number;
  prospect_name: string;
  prospect_position: string;
  prospect_organization: string | null;
  notes: string | null;
  added_at: string;
  notify_on_changes: boolean;
}

export async function addToWatchlist(
  prospectId: number,
  notes?: string,
  notifyOnChanges: boolean = true
): Promise<WatchlistEntry> {
  const response = await apiClient.post<WatchlistEntry>('/watchlist', {
    prospect_id: prospectId,
    notes,
    notify_on_changes: notifyOnChanges,
  });
  return response.data;
}

export async function getWatchlist(): Promise<WatchlistEntry[]> {
  const response = await apiClient.get<WatchlistEntry[]>('/watchlist');
  return response.data;
}

export async function removeFromWatchlist(prospectId: number): Promise<void> {
  await apiClient.delete(`/watchlist/${prospectId}`);
}

export async function updateWatchlistNotes(
  prospectId: number,
  notes: string
): Promise<void> {
  await apiClient.patch(`/watchlist/${prospectId}/notes`, { notes });
}

export async function toggleWatchlistNotifications(
  prospectId: number,
  enabled: boolean
): Promise<void> {
  await apiClient.patch(`/watchlist/${prospectId}/notifications`, { enabled });
}
